### PR TITLE
Add commit new nightly bundle github action.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
## Referenced issue:
https://github.com/eclipse/che/issues/17338

## What does this pull request do:
Add commit new nightly bundle github action: if developer changed deployment yaml files, then github action at the night should commit new bundle and bundle images will be pushed. If there is no changes in the deployment yamls - then nothing to commit.
Fix not needed push bundle images, when we have had already pushed images.

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

**Review by tolusha — APPROVED** [review]

**Comment by AndrienkoAleksandr** [comment]
[retest]

File: .github/workflows/pushCatalogImages.yaml [hunk 1]
```diff
@@ -19,10 +19,28 @@ jobs:
...
+        fetch-depth: 0
+
+    - name: Prepare new nightly Olm bundle
+      run: >-
+        export GOPATH="${GITHUB_WORKSPACE}/go" &&
+        echo "GOPATH: ${GOPATH}" &&
+        export GITHUB_WORKSPACE="${GOPATH}/src/github.com/eclipse/${{ github.repository }}" &&
+        cd ${GITHUB_WORKSPACE} && "${GITHUB_WORKSPACE}/.github/action_scripts/provideNewNightlyBundle.sh"
+
+    - name: Commit new nightly Olm bundle
```
Comment (flacatus) [inline .github/workflows/pushCatalogImages.yaml:37]: WDYT about adding a custom message in the commit?

Comment (AndrienkoAleksandr) [reply .github/workflows/pushCatalogImages.yaml:37]: It's a good idea, but I think, we can improve it in the next pr's.

**Pull request merged**